app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,48 @@
+require 'date'
+
+class SalesController < ApplicationController
+
+  def show
+    @sale = Sale.where(vendor_id: params[:id]).order(purchase_time: :asc)
+    @group = Sale.group("strftime('%m', purchase_time)").count
```

Comment:
It looks like this is finding the number of sales in a given month? This would be good as a method in the Sale model (models/sale.rb).
```def num_sales_by_month(month)
...
end```